test/jdk/javax/swing/JPopupMenu/MouseDragPopupTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JPopupMenu;
+import javax.swing.SwingUtilities;
+import javax.swing.event.PopupMenuEvent;
+import javax.swing.event.PopupMenuListener;
+
+/*
+ * @test
+ * @bug 8315655
+ * @key headful
+ * @run main MouseDragPopupTest
+ */
+public class MouseDragPopupTest {
+    static volatile boolean failed;
+    static volatile JFrame frame;
+    static volatile JPanel panel;
+    static volatile Point srcPoint;
+    static volatile Dimension d;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            failed = false;
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+
+            SwingUtilities.invokeAndWait(() -> {
+                createAndShowGUI();
+            });
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            SwingUtilities.invokeAndWait(() -> {
+                srcPoint = frame.getLocationOnScreen();
+                d = frame.getSize();
+            });
+            robot.waitForIdle();
+            srcPoint.translate(2 * d.width / 3, 3 * d.height / 4);
+
+            final Point dstPoint = new Point(srcPoint);
+            dstPoint.translate(4 * d.width / 15, 0);
+
+
+            robot.mouseMove(srcPoint.x, srcPoint.y);
+            robot.waitForIdle();
+
+            robot.mousePress(InputEvent.BUTTON3_DOWN_MASK);
+            robot.waitForIdle();
+
+            while (!srcPoint.equals(dstPoint)) {
+                srcPoint.translate(sign(dstPoint.x - srcPoint.x), 0);
+                robot.mouseMove(srcPoint.x, srcPoint.y);
+            }
+            robot.waitForIdle();
+
+            robot.mouseRelease(InputEvent.BUTTON3_DOWN_MASK);
```

Comment:
Better to release in `finally` block so that if unexpectedly any exception occurs it will not hamper any other test if tun as part of CI job..